packages/salesforcedx-vscode-soql/src/editor/soqlEditorProvider.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+export class SOQLEditorProvider implements vscode.CustomTextEditorProvider {
+  public static register(context: vscode.ExtensionContext) {
+    const provider = new SOQLEditorProvider(context);
+    const providerRegistration = vscode.window.registerCustomEditorProvider(
+      SOQLEditorProvider.viewType,
+      provider
+    );
+    return providerRegistration;
+  }
+
+  private static readonly viewType = 'soqlCustom.soql';
+  constructor(private readonly context: vscode.ExtensionContext) {}
+
+  public async resolveCustomTextEditor(
+    document: vscode.TextDocument,
+    webviewPanel: vscode.WebviewPanel,
+    _token: vscode.CancellationToken
+  ): Promise<void> {
+    webviewPanel.webview.options = {
+      enableScripts: true,
+      localResourceRoots: [
+        vscode.Uri.file(
+          path.join(
+            this.context.extensionPath,
+            'node_modules/@salesforce/soql-builder-ui'
+          )
+        )
+      ]
+    };
+
+    webviewPanel.webview.html = this.getWebViewContent(webviewPanel.webview);
+  }
+
+  private getWebViewContent(webview: vscode.Webview): string {
+    const pathToLwcDist = path.join(
+      this.context.extensionPath,
+      'node_modules/@salesforce/soql-builder-ui/dist'
+    );
+    const pathToHtml = path.join(pathToLwcDist, 'index.html');
+    let html = fs.readFileSync(pathToHtml).toString();
+    const scriptUri = webview.asWebviewUri(
+      vscode.Uri.file(path.join(pathToLwcDist, 'app.js'))
+    );
+    const zeroDotScriptUri = webview.asWebviewUri(
+      vscode.Uri.file(path.join(pathToLwcDist, '0.app.js'))
+    );
+    const cspMetaTag: string = `<meta
+      http-equiv="Content-Security-Policy"
+      content="default-src 'none';
+      img-src ${webview.cspSource} https:;
+      script-src ${webview.cspSource};
+      style-src 'unsafe-inline' ${webview.cspSource};"
+    />`;
+
+    html = html.replace('<!-- CSP TAG -->', cspMetaTag);
```

Comment:
@xyc I went a slightly different direction with the implementation, but this will accomplish what we are looking for and solve issues for us in the other tooling repo with local development and the CSP.